src/Features/CSharp/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.cs

```diff
@@ -1,332 +1,123 @@
-                    binaryExpression.Left,
-                    (LiteralExpressionSyntax)binaryExpression.Right,
-                    semanticModel,
-                    cancellationToken);
+                newIfNodeStatement = ifNode.Else.Statement.Kind() != SyntaxKind.Block
+                    ? SyntaxFactory.Block(ifNode.Else.Statement)
+                    : ifNode.Else.Statement;
+                newElseStatement = ifNode.Else.WithStatement(ifNode.Statement);
             }
-            else if (binaryExpression.Kind() == SyntaxKind.EqualsExpression &&
-                     binaryExpression.Left.Kind() == SyntaxKind.NumericLiteralExpression)
+            else
             {
-                canSimplify = CanSimplifyToLengthEqualsZeroExpression(
-                    binaryExpression.Right,
-                    (LiteralExpressionSyntax)binaryExpression.Left,
-                    semanticModel,
-                    cancellationToken);
+                var elseTrailingTrivia = ifNode.Else.GetTrailingTrivia();
+                var ifTrailingTrivia = ifNode.Statement.GetTrailingTrivia();
+                newIfNodeStatement = ifNode.Else.Statement.WithTrailingTrivia(ifTrailingTrivia);
+                newElseStatement = ifNode.Else.WithStatement(ifNode.Statement).WithTrailingTrivia(elseTrailingTrivia);
             }
 
-            return canSimplify;
-        }
-
-        private bool CanSimplifyToLengthEqualsZeroExpression(
-            ExpressionSyntax variableExpression,
-            LiteralExpressionSyntax numericLiteralExpression,
-            SemanticModel semanticModel,
-            CancellationToken cancellationToken)
-        {
-            var numericValue = semanticModel.GetConstantValue(numericLiteralExpression, cancellationToken);
-            if (numericValue.HasValue && numericValue.Value is int && (int)numericValue.Value == 0)
-            {
-                var symbol = semanticModel.GetSymbolInfo(variableExpression, cancellationToken).Symbol;
+            var newIfStatment = ifNode.Else.Statement.Kind() == SyntaxKind.IfStatement && newIfNodeStatement.Kind() != SyntaxKind.Block
+                        ? SyntaxFactory.Block(newIfNodeStatement)
+                        : newIfNodeStatement;
 
-                if (symbol != null && (symbol.Name == "Length" || symbol.Name == "LongLength"))
-                {
-                    var containingType = symbol.ContainingType;
-                    if (containingType != null &&
-                        (containingType.SpecialType == SpecialType.System_Array ||
-                         containingType.SpecialType == SpecialType.System_String))
-                    {
-                        return true;
-                    }
-                }
+            ifNode = ifNode.WithCondition((ExpressionSyntax)(Negate(ifNode.Condition, generator, syntaxFacts, model, cancellationToken)))
+                .WithStatement(newIfStatment)
+                .WithElse(newElseStatement);
```

Comment:
see comments below about how you can do this without needing all the explicit casts here.